lib/Cron/Workers/WooCommerceSync.php

```diff
@@ -29,22 +29,23 @@ function checkProcessingRequirements() {
     return $this->woocommerce_helper->isWooCommerceActive();
   }
 
-  function prepareTask(ScheduledTask $task) {
-    if (is_null($task->status)) {
-      $this->taskAlreadyInProgress = true;
-      return false;
-    }
-
-    return parent::prepareTask($task);
-  }
-
   function processTaskStrategy(ScheduledTask $task) {
-    if ($this->taskAlreadyInProgress) {
+    $meta = $task->getMeta();
+    if (!empty($meta['in_progress'])) {
```

Comment:
Do we need extra meta field for this information? I think that the same information is stored in `status` (`$task->status === null`), isn't it?

Reply:
Not quite. The `processTask` method is called on tasks with `status = null` on every cron run. We can't distinguish between a first and non-first run by the status itself, so I've added a flag for this purpose.

Reply:
Ahh I see. I always thought that prepareTask and porcessTask are called within one run :(

Reply:
Yeah, I forgot this too when I did the first version of the fix. This was done because some tasks require expensive preparation (e.g. populating subscriber queues).